src/electionguard/ballot.py

```diff
@@ -261,8 +232,59 @@ def crypto_hash_with(self, seed_hash: ElementModQ) -> ElementModQ:
 
         In most cases the seed_hash should match the `description_hash`
         """
+        return _ciphertext_ballot_selection_crypto_hash_with(
+            self.object_id, seed_hash, self.message
+        )
+
 
-        return hash_elems(self.object_id, seed_hash, self.message.crypto_hash())
+def _ciphertext_ballot_selection_crypto_hash_with(
+    object_id: str, seed_hash: ElementModQ, message: ElGamalCiphertext
+) -> ElementModQ:
+    return hash_elems(object_id, seed_hash, message.crypto_hash())
+
+
+def make_ciphertext_ballot_selection(
+    object_id: str,
+    description_hash: ElementModQ,
+    message: ElGamalCiphertext,
+    elgamal_public_key: ElementModP,
+    proof_seed: ElementModQ,
+    selection_representation: int,
+    is_placeholder_selection: bool = False,
+    nonce: Optional[ElementModQ] = None,
+    crypto_hash: Optional[ElementModQ] = None,
+    proof: Optional[DisjunctiveChaumPedersenProof] = None,
+    extended_data: Optional[ElGamalCiphertext] = None,
+) -> CiphertextBallotSelection:
+    """
+    Constructs a `CipherTextBallotSelection` object. Most of the parameters here match up to fields
+    in the class, but this helper function will optionally compute a Chaum-Pedersen proof if the
+    given nonce isn't `None`. Likewise, if a crypto_hash is not provided, it will be derived from
+    the other fields.
```

Comment:
i think we should call out each parameter in the summary description since there are so many

Reply:
I want to emphasize this. I'm still only 50% sure what the different sorts of hashes are. So long as you're trying to make these things more clear, coming up with better nomenclature for the various hashes floating around the system sounds really important.

Reply:
The better naming on the hashes was something also mentioned by Josh about the output data and should probably be in PR #114 . The issue is that there weren't a lot of good names for these in the first place.